include/flux/op/find.hpp

```diff
@@ -6,21 +6,58 @@
 #ifndef FLUX_OP_FIND_HPP_INCLUDED
 #define FLUX_OP_FIND_HPP_INCLUDED
 
+#include <flux/core/utils.hpp>
 #include <flux/op/for_each_while.hpp>
 
+#include <cstring>
+#include <type_traits>
+
 namespace flux {
 
 namespace detail {
 
 struct find_fn {
-    template <sequence Seq, typename Value>
-        requires std::equality_comparable_with<element_t<Seq>, Value const&>
-    constexpr auto operator()(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+private:
+    template <typename Seq, typename Value>
+    static constexpr auto impl(Seq&& seq, Value const& value) -> cursor_t<Seq>
     {
         return for_each_while(seq, [&](auto&& elem) {
             return FLUX_FWD(elem) != value;
         });
     }
+
+public:
+    template <sequence Seq, typename Value>
+        requires std::equality_comparable_with<element_t<Seq>, Value const&>
+    constexpr auto operator()(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+    {
+        constexpr auto can_memchr = 
+            contiguous_sequence<Seq> && sized_sequence<Seq> && 
+            std::same_as<Value, value_t<Seq>> &&
+            flux::detail::any_of<value_t<Seq>, char, signed char, unsigned char, char8_t, std::byte>;
+
+        if constexpr (can_memchr) {
+            if (std::is_constant_evaluated()) {
+                return impl(seq, value);
+            } else {
+                auto size = flux::usize(seq);
+                if(size == 0) {
+                    return flux::last(seq);
+                }
+                FLUX_ASSERT(flux::data(seq) != nullptr);
+                auto location = std::memchr(flux::data(seq), static_cast<unsigned char>(value),
+                    flux::usize(seq) * sizeof(value_t<Seq>));
```

Comment:
Again, we can re-use the `size` variable

```suggestion
                auto location = std::memchr(flux::data(seq), static_cast<unsigned char>(value),
                    size * sizeof(value_t<Seq>));
```